### PR TITLE
[netatmo] Floodlight switch repaired

### LINKED ISSUE #12741: [netatmo] The floodlight isn't working anymore
The API to change the floodlight was changed (and is now also officially documentated). Therefore it isn't possible anymore to switch the floodlight of the outdoor camera (to on, off or auto).

The current version of the binding (OpenHAB 3.3.0 M5) is also affected.

### PR DESCRIPTION
**Pull request by Novanic**
Signed-off-by: Sven Strohschein <sven.strohschein@gmail.com>

Pull request to fix the floodlight switch modes for the outdoor camera. See issue #12741 for more information.

Fixes #12741

### CODE REVIEW

**Review by lolodomo — APPROVED** [review]
Code looks good to me.

**Comment by lolodomo** [comment]
I let @clinique have a look

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/SecurityApi.java [hunk 1]
```diff
@@ -101,10 +101,13 @@ public void changeStatus(String localCameraURL, boolean setOn) throws NetatmoExc
         post(uriBuilder, ApiResponse.Ok.class, null, null);
     }
 
-    public void changeFloodLightMode(String localCameraURL, FloodLightMode mode) throws NetatmoException {
-        UriBuilder uriBuilder = UriBuilder.fromUri(localCameraURL).path(PATH_COMMAND).path(SUB_PATH_FLOODLIGHTSET);
+    public void changeFloodLightMode(String homeId, String cameraId, FloodLightMode mode) throws NetatmoException {
+        UriBuilder uriBuilder = getAppUriBuilder(PATH_STATE);
         uriBuilder.queryParam("config", "%7B%22mode%22:%22" + mode.toString() + "%22%7D");
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/SecurityApi.java:106]: Is this queryParam still needed now that you send a payload with the target mode in it ?

**Review by clinique — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/SecurityApi.java (hunk 1 above)
Comment (Novanic) [reply bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/SecurityApi.java]: You are right, that query parameter isn't required anymore. It is now removed.

**Review by clinique — APPROVED** [review]

**Pull request merged**